x/concentrated-liquidity/swaps.go

```diff
@@ -309,8 +316,8 @@ func (k Keeper) computeOutAmtGivenIn(
 	// initialize swap state with the following parameters:
 	// as we iterate through the following for loop, this swap state will get updated after each required iteration
 	swapState := SwapState{
-		amountSpecifiedRemaining: tokenAmountInSpecified, // tokenIn
-		amountCalculated:         sdk.ZeroDec(),          // tokenOut
+		amountSpecifiedRemaining: tokenAmountInSpecified.Clone(), // tokenIn
```

Comment:
I see you used clone here, but for computeIn in the same step you did not. Just pointing this out

Reply:
IMO we do not need to use clone here

Reply:
if its remotely concerning, we can definitely keep cloning

(An extra heap allocation is better for confidence in 0 chance of bugs)

I'm supportive of doing clone to be more future proof here. The other changes are pretty trivial / isolated to convince yourself about, this one not as much

Reply:
Added comment:
```
// N.B. We clone tokenAmountInSpecified because swapState.amountSpecifiedRemaining will get
// mutated during the compute swap step loop. However, we still need the original
// value of tokenAmountInSpecified to calculate the amount of tokenOut to return.
```

This does not apply to in given out. That's why we don't clone there